testcases/crypto/ec_func.c

```diff
@@ -162,6 +162,566 @@ _signVerifyParam signVerifyInput[] = {
+
+        CK_ATTRIBUTE pub_attr[] = {
+            {CKA_ECDSA_PARAMS, der_ec_supported[i].curve, der_ec_supported[i].size},
+            {CKA_VERIFY, &true, sizeof(true)},
+            {CKA_MODIFIABLE, &true, sizeof(true)},
+        };
+        CK_ULONG pub_attr_len = sizeof(pub_attr)/sizeof(CK_ATTRIBUTE);
+
+        CK_ATTRIBUTE  extr1_tmpl[] = {
+            {CKA_EC_POINT, pubkeyA_value, sizeof(pubkeyA_value)},
+        };
+        CK_ULONG extr1_tmpl_len = sizeof(extr1_tmpl)/sizeof(CK_ATTRIBUTE);
+
+        CK_ATTRIBUTE  extr2_tmpl[] = {
+            {CKA_EC_POINT, pubkeyB_value, sizeof(pubkeyB_value)},
+        };
+        CK_ULONG extr2_tmpl_len = sizeof(extr2_tmpl)/sizeof(CK_ATTRIBUTE);
+
+        if (!(is_cca_token(SLOT_ID))) {
+            if (!memcmp(der_ec_supported[i].curve, brainpoolP512t1, sizeof(brainpoolP512t1))) {
+                testcase_skip("Slot %u doesn't support this curve", (unsigned int)SLOT_ID);
+                continue;
+            }
+        }
+
+        // Testcase #1 - Generate 2 EC key pairs.
+
+        // First, generate the EC key pair for party A
+        mech.mechanism = CKM_EC_KEY_PAIR_GEN;
+        mech.ulParameterLen = 0;
+        mech.pParameter = NULL;
+
+        rc = funcs->C_GenerateKeyPair(session, &mech, pub_attr, pub_attr_len, prv_attr, prv_attr_len, &publ_keyA, &priv_keyA);
+        if (rc != CKR_OK) {
+            testcase_fail("C_GenerateKeyPair with valid input failed at i=%lu, rc=%s", i, p11_get_ckr(rc));
+            goto testcase_cleanup;
+        }
+
+        // Extract public key A
+        rc = funcs->C_GetAttributeValue(session, publ_keyA, extr1_tmpl, extr1_tmpl_len);
+        if (rc != CKR_OK) {
+            testcase_error("C_GetAttributeValue #1: rc = %s", p11_get_ckr(rc));
+            goto testcase_cleanup;
+        }
+
+        // Now generate the EC key pair for party B
+        mech.mechanism = CKM_EC_KEY_PAIR_GEN;
+        mech.ulParameterLen = 0;
+        mech.pParameter = NULL;
+
+        rc = funcs->C_GenerateKeyPair(session, &mech, pub_attr, pub_attr_len, prv_attr, prv_attr_len, &publ_keyB, &priv_keyB);
+        if (rc != CKR_OK) {
+            testcase_fail("C_GenerateKeyPair with valid input failed at i=%lu, rc=%s", i, p11_get_ckr(rc));
+            goto testcase_cleanup;
+        }
+
+        // Extract public key B
+        rc = funcs->C_GetAttributeValue(session, publ_keyB, extr2_tmpl, extr2_tmpl_len);
+        if (rc != CKR_OK) {
+            testcase_error("C_GetAttributeValue #1: rc = %s", p11_get_ckr(rc));
+            goto testcase_cleanup;
+        }
+
+        // Check if the key lengths are equal
+        if (extr1_tmpl->ulValueLen != extr2_tmpl->ulValueLen) {
+            testcase_error("Length of public key A not equal to length of public key B");
+            goto testcase_cleanup;
+        }
+
+        // Testcase #2 - Now derive the secrets...
+
+        for (int j=0; j < NUM_KDFS; j++) {
+
+            for (int k=0; k<NUM_SECRET_KEY_LENGTHS; k++) {
```

Comment:
Do not declare int k inside the for, please do it outside. This is causing Travis CI build to fail.